modules/aws/lls/lls-startup.sh.tmpl

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# Copyright (c) 2020 Teradici Corporation
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
+
+INST_LOG_FILE="/var/log/teradici/lls/install.log"
+LLS_YUM_PKG="pcoip-license-server"
+
+log() {
+    local message="$1"
+    echo "[$(date)] $${message}" | tee -a "$INST_LOG_FILE"
+}
+
+get_credentials() {
+    if [[ -z "${customer_master_key_id}" ]]
+    then
+        log "Not using encryption"
+
+        LLS_ADMIN_PASSWORD=${lls_admin_password}
+        LLS_ACTIVATION_CODE=${lls_activation_code}
+
+    else
+        log "Using encryption key ${customer_master_key_id}"
+
+        LLS_ADMIN_PASSWORD=$(aws kms decrypt --region ${aws_region} --ciphertext-blob fileb://<(echo "${lls_admin_password}" | base64 -d) --output text --query Plaintext | base64 -d)
+        LL_ACTIVATION_CODE=$(aws kms decrypt --region ${aws_region} --ciphertext-blob fileb://<(echo "${lls_activation_code}" | base64 -d) --output text --query Plaintext | base64 -d)
+    fi
+
+    # Exit if any of the required variables are missing
+    if [[ -z "$LLS_ADMIN_PASSWORD" || -z "$LLS_ACTIVATION_CODE" ]]
+    then
+        log "Missing required parameters:"
+        exit 1
+    fi
+}
+
+if [[ ! -f "$INST_LOG_FILE" ]]
+then
+    mkdir -p "$(dirname $${INST_LOG_FILE})"
+    touch "$INST_LOG_FILE"
+    chmod 600 "$INST_LOG_FILE"
+fi
+
+yum info $LLS_YUM_PKG
+if [[ $? -eq 0 ]]
+then
+    log "PCoIP licence Server already installed. Skipping startup script."
+    exit 0
+fi
+
+log "$(date)"
+
+yum update -y
+yum install -y wget yum-utils
+
+get_credentials
+
+yum install -y ${lls_repo_url}
+yum install -y $LLS_YUM_PKG
+
+pcoip-set-password -p "1P@ssw0rd!" -n "$LLS_ADMIN_PASSWORD"
```

Comment:
What's this password for?

Reply:
when installed, lls has a default password. we need to change it to a new password that the user supplies.